specs/_features/ePBS/builder.md

```diff
@@ -0,0 +1,229 @@
+**Table of Contents**  *generated with [DocToc](https://github.com/thlorenz/doctoc)*
+
+- [ePBS -- Honest Builder](#epbs----honest-builder)
+  - [Introduction](#introduction)
+  - [Becoming a builder](#becoming-a-builder)
+    - [Initialization](#initialization)
+        - [Builder withdrawal credential](#builder-withdrawal-credential)
+    - [Submit deposit](#submit-deposit)
+    - [Process deposit](#process-deposit)
+    - [Activation](#activation)
+  - [Beacon chain responsibilities](#beacon-chain-responsibilities)
+    - [Signed execution payload header envelope proposal](#signed-execution-payload-header-envelope-proposal)
+      - [Constructing the `SignedExecutionPayloadHeaderEnvelope`](#constructing-the-signedexecutionpayloadheaderenvelope)
+      - [Broadcast execution payload header envelope](#broadcast-execution-payload-header-envelope)
+    - [Signed execution payload envelope construction](#signed-execution-payload-envelope-construction)
+      - [Determine if it is safe to reveal](#determine-if-it-is-safe-to-reveal)
+      - [Constructing the `SignedExecutionPayloadEnvelope`](#constructing-the-signedexecutionpayloadenvelope)
+      - [Broadcast execution payload envelope](#broadcast-execution-payload-envelope)
+      - [Broadcast blob sidecars](#broadcast-blob-sidecars)
+  - [Design Decision Rationale](#design-decision-rationale)
+    - [What are the changes with regard to blob sidecars?](#what-are-the-changes-with-regard-to-blob-sidecars)
+    - [How does builder to proposer payment work?](#how-does-builder-to-proposer-payment-work)
+    - [Should the builder worry about same slot unbundling?](#should-the-builder-worry-about-same-slot-unbundling)
+    - [What if builder sees proposer equivocate?](#what-if-builder-sees-proposer-equivocate)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# ePBS -- Honest Builder
+
+**Notice**: This document is a work-in-progress for researchers and implementers.
+
+
+## Introduction
+
+This document represents the expected behavior of an "honest builder" with respect to ePBS of the Ethereum proof-of-stake protocol. 
+This document does not distinguish between a "node" (i.e. the functionality of following and reading the beacon chain) and a "builder" (i.e. the functionality of actively participating in consensus).
+The separation of concerns between these (potentially) two pieces of software is left as a design decision that is out of scope.
+
+A builder is an entity that participates in the block building of the Ethereum proof-of-stake protocol. This is an optional role for users in which they can post ETH as collateral and build blocks to see financial returns in exchange for building the protocol.
+In this design, builder is an extension of validator using a different withdrawal prefix `0x0b`. The protocol onboards builder by upgrading validators into builders if minimal staked balance is reached.
+
+## Becoming a builder
```

Comment:
A builder must keep a minimum balance of `BUILDER_MIN_BALANCE` in order to continue performing building duties.